py/formatfloat.c

```diff
@@ -96,7 +96,21 @@ static inline int fp_isless1(float x) {
 #define fp_iszero(x) (x == 0)
 #define fp_isless1(x) (x < 1.0)
 
-#endif
+union floatbits {
+    double f;
+    uint64_t u;
+};
```

Comment:
I think you can use the existing `mp_float_union_t` instead of needed to add this union definition.

Reply:
I could, but then I'd have to have different implementations of ```fp_ge_eps`` for float32 and float64 conditions.  I think it's more legible to have parallel types and a a single ``fp_ge_eps``.  Rewriting the float32 special-case functions for an unsigned int member is awkward.

Reply:
Oh, I didn't realise the `MICROPY_FLOAT_IMPL_FLOAT` case just above also defines `union floatbits`.  In that case `mp_float_union_t` is exactly what you need, it's defined in `py/misc.h` for both the float and double cases.